### DISCUSSION TITLE
Add ServiceStack reference get an error in Dart App

### DISCUSSION THREAD

**Discussion by mylvghb** [ServiceStack/Discuss#91] [Q&A] [answered]
When I add an SS reference in a console-full app of dart-grpc using the command `x proto-dart https://localhost:5051 -out lib`, I get an error on the server side: HTTP/2 over TLS was not negotiated on an HTTP/2- only endpoint.

Server:

![image](https://github.com/ServiceStack/Discuss/assets/35945793/8678f661-653c-4ac9-8117-84dd06681c06)

Dart Client:

![image](https://github.com/ServiceStack/Discuss/assets/35945793/76d42785-48f3-4a89-a3d9-d9424157a2d6)

How to fix it?

**Comment by Layoric** [comment, marked as answer]
You will want to use the `x` tool with the normal HTTP host running locally, eg 
```
x proto-dart https://localhost:5001 -out lib
```

**Reply by mylvghb** [reply]
That's the command I used.

**Reply by Layoric** [reply]
In your screenshot it shows you using `https://localhost:5051`. Note the difference in the port used.
